packages/opencensus-web-exporter-ocagent/src/adapters.ts

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as coreTypes from '@opencensus/core';
+import * as apiTypes from './api-types';
+
+/**
+ * Converts a RootSpan type from @opencensus/core to the Span JSON structure
+ * expected by the OpenCensus Agent's HTTP/JSON (grpc-gateway) API.
+ */
+export function adaptRootSpan(rootSpan: coreTypes.RootSpan): apiTypes.Span[] {
+  const adaptedSpans: apiTypes.Span[] = rootSpan.spans.map(adaptSpan);
+  adaptedSpans.unshift(adaptSpan(rootSpan));
+  return adaptedSpans;
+}
+
+function adaptString(value: string): apiTypes.TruncatableString {
+  return {value};
+}
+
+/** Converts hexadecimal string to base64 string. */
+function hexToBase64(hexStr: string): string {
+  const hexStrLen = hexStr.length;
+  let hexAsciiCharsStr = '';
+  for (let i = 0; i < hexStrLen; i += 2) {
+    const hexPair = hexStr.substring(i, i + 2);
+    const hexVal = parseInt(hexPair, 16);
+    hexAsciiCharsStr += String.fromCharCode(hexVal);
+  }
+  return btoa(hexAsciiCharsStr);
+}
+
+function adaptTraceState(coreTraceState?: coreTypes.TraceState):
+    apiTypes.TraceState {
+  if (!coreTraceState || !coreTraceState.length) return {};
+  const entries = coreTraceState.split(',');
+  const apiTraceState: apiTypes.TraceState = {};
+  for (const entry of entries) {
+    const [key, value] = entry.split('=');
+    apiTraceState[key] = value;
+  }
+  return apiTraceState;
+}
+
+function adaptSpanKind(coreKind: string): apiTypes.SpanKind {
+  switch (coreKind) {
+    case 'SERVER': {
+      return apiTypes.SpanKind.SERVER;
+    }
+    case 'CLIENT': {
+      return apiTypes.SpanKind.CLIENT;
+    }
+    default: {
+      return apiTypes.SpanKind.UNSPECIFIED;
+    }
+  }
+}
+
+function adaptValue(value: boolean|string|number): apiTypes.AttributeValue {
+  const valType = typeof value;
+  if (valType === 'boolean') {
+    return {boolValue: value as boolean};
+  }
+  if (valType === 'number' && value === Math.floor(value as number)) {
+    return {intValue: String(value)};
```

Comment:
Good point - the `intValue` would be written as a string from a read API that was based on int64 protobuf values because the JS `number` can't accurately hold all 64-bit ints.

But here we are going the other way so we are fine to keep it as a number and I tested with the OpenCensus agent grpc-gateway and it works fine as a number.

I noticed now that there is support for a `doubleValue` attribute type since https://github.com/census-instrumentation/opencensus-proto/pull/98/files . That seems like a more exact match with the JS `number` type so I changed it to that here and in the tests. Maybe if this eventually supports big ints it could use `intValue`, but full browser support for that is still a while off.